### PR TITLE
[BUDI-8091] Fix renaming columns in SQL Server (and MySQL)

### PR DESCRIPTION
**Pull request by samwho**
## Description

This PR addresses a bug users would see when trying to rename columns in an SQL Server external datasource. To attack this problem I first went to write a test, and realised that the `table.spec.ts` test suite was only hitting internal datasources. I modified the suite to hit our supported external datasources for testing, which revealed the SQL Server bug and also a different bug in MySQL.

The problem was that the query `knex` generates doesn't appear to be correct. It tries to use a parameterised query that looks like this:

```sql
exec sp_rename ?, ?, 'COLUMN';
```

We've "fixed" this by not using parameters in this query.

The MySQL bug was slightly different. It appears that sometimes the query to rename a MySQL column doesn't have the column type attached to it, and the query it tries to run relies on knowing the type:

```sql
alter table `table` change column `name` `first_name` varchar(45);
```

Fixed this by removing the need to know the column type at all:

```sql
alter table `table` rename column `name` to `first_name`;
```

## Addresses
- https://linear.app/budibase/issue/BUDI-8091/cant-rename-ms-sql-columns-incorrect-syntax-near

## Launchcontrol

Fixes a bug that prevented you from renaming columns in an SQL Server external datasource, and a more subtle bug that would prevent you renaming columns in MySQL external datasources under some circumstances.

### CODE REVIEW

File: packages/server/src/api/routes/tests/table.spec.ts [hunk 1]
```diff
@@ -1,146 +1,82 @@
...
+import { tableForDatasource } from "../../../tests/utilities/structures"
+import timekeeper from "timekeeper"
 
 const { basicTable } = setup.structures
 const ISO_REGEX_PATTERN = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/
 
-describe("/tables", () => {
-  let request = setup.getRequest()
+describe.each([
+  // ["internal", undefined],
```
[the remark this answers is not in the document]
Comment (samwho) [reply packages/server/src/api/routes/tests/table.spec.ts]: I don't wanna talk about it.

File: packages/server/src/integrations/base/sqlTable.ts [hunk 2]
```diff
@@ -224,19 +224,40 @@ class SqlTableQueryBuilder {
...
           client,
           json.table,
           json.meta.tables,
           json.meta.table,
           json.meta.renamed!
         )
+
+        // renameColumn for SQL Server returns a parameterised `sp_rename` query,
+        // which is not supported by SQL Server and gives a syntax error.
+        if (this.sqlClient === SqlClient.MS_SQL && json.meta.renamed) {
```
[the remark this answers is not in the document]
Comment (mike12345567) [reply packages/server/src/integrations/base/sqlTable.ts:243]: Also not tied to this, might prove tricky to get MySQL into the same pattern and if it is then I'd just leave it!

**Review by mike12345567 — COMMENTED** [review]
Awesome work on this! The improvements to tests is **massive** - having proper test suites around the table API for external DBs is really a huge addition!

Think its worth investigating if we can combine the MS-SQL and MySQL hack for consistency, but if its not possible, then ready to get this in!

**Review by mike12345567 — APPROVED** [review]

**Comment by samwho** [comment]
Spoke with Michael offline, going to get this in because it fixes a bug. Will explore making it tidier in a follow-up PR.

**Pull request merged**